### PR TITLE
[dev-env] feat: support docker-compose-plugin in addition to docker-compose

### PR DESCRIPTION
**Pull request by cthiounn**
Docker has released a V2 for Docker compose, integrated in Docker CLI, instead of the previous docker-compose binary.

https://docs.docker.com/compose/

By default, Docker suggest to install the docker compose plugin, like in this tutorial : 

https://docs.docker.com/engine/install/ubuntu/

Tournesol's dev_env sh script use the docker-compose binary to set up the dev env. The script can be adapted to automatically detect if that binary is either present or either docker compose plugin is.

enhancements :

* auto detection user's config :
    * A docker-compose is present
    * docker compose plugin is installed
    * None, script should advise to install one
* tear down command

### CODE REVIEW

File: dev-env/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Quick start
 
 ### Requirements:
-* docker-compose
```
Comment (GresilleSiffle) [inline dev-env/README.md:6]:
```suggestion
* docker-compose-plugin (or docker-compose)
```

File: dev-env/run-docker-compose.sh [hunk 2]
```diff
@@ -49,6 +97,13 @@ if [[ "${1:-""}" == 'restart' ]]; then
   exit
 fi
 
+if [[ "${1:-""}" == 'stop' ]]; then
```
Comment (GresilleSiffle) [inline dev-env/run-docker-compose.sh:100]: The commands `down` and `stop` have a different meaning for `compose`.

- `down` stops the services and remove the containers
- `stop` only stops the services

Here the script if answering to a `stop` positional argument with a `down`, which can be confusing.

As we have already a way to recreate the container with the argument `restart` (the `restart` triggers a `docker compose up --force-recreate`), I suggest to simply call `docker compose stop` and rename the related function accordingly.

File: dev-env/run-docker-compose.sh [hunk 3]
```diff
@@ -49,6 +97,13 @@ if [[ "${1:-""}" == 'restart' ]]; then
...
+  echo "Stopping dev containers..."
+  compose_down
```
Comment (GresilleSiffle) [inline dev-env/run-docker-compose.sh:102]: This suggestion also implies to replace all occurrences of  `compose_down` in the script.

```suggestion
  compose_stop
```

**Review by GresilleSiffle — DISMISSED** [review]

**Review by GresilleSiffle — COMMENTED** [review]
Works well on Debian 11, with

- `docker-compose-plugin` version 2.6.0
- and also with `docker-compose` version 1.29.2

**Comment by adrausch** [comment]
Tested on my MacOS and it worked for me

File: dev-env/README.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
...
+* docker-compose (or docker compose plugin)
```
Comment (GresilleSiffle) [inline dev-env/README.md:6]:
```suggestion
* `docker-compose-plugin` (or `docker-compose`)
```

**Review by GresilleSiffle — APPROVED** [review]

**Pull request merged**